src/screen.c

```diff
@@ -2130,6 +2130,19 @@ if (db) fprintf(stderr, "initialize_raw_fb reset\n");
 	if (sscanf(str, "shm:%d", &shmid) == 1) {
 		/* shm:N */
 #if HAVE_XSHM || HAVE_SHMAT
+		/* try to use shm key*/
+		key_t shmkey = 0;
+		size_t size = 0;
+		int newshmid = -1;
+		
+		shmkey = shmid;
+		size = w * h * b / 8;
+		newshmid = shmget(shmkey, size, 0);
```

Comment:
According to the man page, 

>  A new shared memory segment, with size equal to the value of size rounded up to a multiple of PAGE_SIZE, is created if key has the value IPC_PRIVATE or key isn't IPC_PRIVATE,  no
       shared memory segment corresponding to key exists, and IPC_CREAT is specified in shmflg.

So this likely breaks the original behaviour where the user enters some shmid and expects the shmat() down below to attach this. With this change, it's very likely that a new shared memory segment is created; this is not what the users expect.

Reply:
Oh, I make a mistake. For an existed shared memory, the size for shmget should be 0. Then this will not alloc new shared memory segment and use the old way to seek shmid. 
The change has committed, thanks for your review!